toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1440,3 +1575,48 @@ func findAllRPMS(rpmsToFind []string) (foundAllRpms bool, missingRpms []string)
 
 	return
 }
+
+// findAllRPMSUpstream returns true if all RPMs requested are found in PMC/repo.
+//  This is used to check if all RPMs are available in PMC/repo before starting a build.
+//  needs repo-list, manifest files rpmrepocloner
+func findAllRPMSUpstream(rpmsToFind []*PkgNode, outDir, tmpDir, workerTar, existingRpmsDir, existingToolchainRpmsDir string, usePreviewRepo, ignoreVersionToResolveSelfDep bool, repoFiles []string) (foundAllRpms bool) {
+	cloner := rpmrepocloner.New()
+	err := cloner.Initialize(outDir, tmpDir, workerTar, existingRpmsDir, existingToolchainRpmsDir, usePreviewRepo, false, repoFiles)
+	if err != nil {
+		logger.Log.Errorf("Failed to initialize RPM repo cloner. Error: %s", err)
+		return
+	}
+	defer cloner.Close()
+
+	err = cloner.AddNetworkFiles("", "")
+	if err != nil {
+		logger.Log.Errorf("Failed to customize RPM repo cloner. Error: %s", err)
+	}
+
```

Comment:
Recent update: this cloner init can now be replaced with `ConstructClonerWithNetwork`.

Reply:
Sorry for the churn, per request I'm going to rename this function to `ConstructCloner()`: https://github.com/microsoft/CBL-Mariner/pull/5595/commits/36ad1cb22a4d9fb5b984f9e909510c00743b7dd7

(Just a heads up, depending on sequencing one of us will have to fix this)